### PR TITLE
[BUGFIX beta] Fixes `computed.sort works incorrectly with empty sort properties array`

### PR DESCRIPTION
**Pull request by onechiporenko**
Fix for #13037

### CODE REVIEW

**Comment by mmun** [comment]
Looks good to me. Is this a regression? We may want to [BUGFIX release] :/

File: packages/ember-runtime/lib/computed/reduce_computed_macros.js [hunk 1]
```diff
@@ -668,6 +668,9 @@ function normalizeSortProperties(sortProperties) {
 }
 
 function sortByNormalizedSortProperties(items, normalizedSortProperties) {
+  if (normalizedSortProperties.length === 0) {
```
Comment (stefanpenner) [inline packages/ember-runtime/lib/computed/reduce_computed_macros.js]: lets do.

```js
let items = emberA(_items.slice());
if (normalizedSortProperties.length === 0) {
  return items;
} 

....
```

Comment (stefanpenner) [inline packages/ember-runtime/lib/computed/reduce_computed_macros.js]: lets do
```js
function ...(_items, ...) {
let items = _items.slice()
...

}
```

**Comment by stefanpenner** [comment]
one nit-pick, otherwise I also think this looks good

**Comment by stefanpenner** [comment]
after reading https://github.com/emberjs/ember.js/issues/13037#issuecomment-198420476 I think this needs more thought.

File: packages/ember-runtime/lib/computed/reduce_computed_macros.js (hunk 1 above)
Comment (stefanpenner) [inline packages/ember-runtime/lib/computed/reduce_computed_macros.js]: wait shouldn't we still cal sort on items, just without a comparator?

**Comment by stefanpenner** [comment]
more discussion: https://github.com/emberjs/ember.js/issues/13037#issuecomment-198436803

I think we need someone to check what the 1.13.x behavior was

**Comment by stefanpenner** [comment]
After discussion, I believe the logic we want is as follows:

```js
Ember.A(Ember.isEmpty(sortProperties) ? inputArray.slice() : sort(inputArray));
```

@mmun r?

**Comment by mmun** [comment]
:+1:

For future proofing we should assert that sortProperties is not a string or non-null object or something.

**Comment by stefanpenner** [comment]
@mmun it seems like null/undefined is fine no? Just means no sort

**Comment by onechiporenko** [comment]
Any updates?

**Comment by rox163** [comment]
Any update on this? We hit this sort issue when we upgraded our app to Ember 2.4 from 2.3

**Comment by mmun** [comment]
@onechiporenko Sorry for not following up. Do you mind rebasing? or I can if you don't have time.

**Comment by onechiporenko** [comment]
@mmun, I would be grateful if you do.
Thanks

**Pull request closed without merging**